utils/cups-browsed.c

```diff
@@ -10712,6 +10713,12 @@ browse_poll_get_printers (browsepoll_t *context, http_t *conn)
       found_cups_printer (context->server, uri, location, info);
       printer = new_browsepoll_printer (uri, location, info);
       printers = g_list_insert (printers, printer, 0);
+      if (NewBrowsePollQueuesShared) {
+      int num_options = 0;
+	    num_options = cupsAddOption("printer-is-shared", "true",
+				    num_options, NULL);
+	    debug_printf("Setting printer-is-shared bit.\n");
+      }
     }
```

Comment:
To which option set are you adding the "printer-is-shared" option here? "cupsAddOption()" needs the option set to add the option to as its 4th argument, where you set NULL. See the source code of cupsAddOption() in CUPS, file `cups/options.c`. Did you actually build and test this code?

Reply:
@tillkamppeter , I added this commit to get reviews from you about the code and therefore, did not completely test the code. However, I have added a new commit, after studying more about sending requests and changing options, which would change the `printer-is-shared` option when required.

Reply:
This also is not the correct thing. Here you are setting the printer-is-shared bit on the remote print queue. Once this does not make sense as on the remote server all these printers are shared, otherwise you would not see them as a client. Also you are sending the request for setting the shared bit to your local CUPS, but the printers are on remote CUPS servers. The request will error.
Did you test what you are doing? With the new option set you should have your local queues for BrowsePolled printers shared, with the new option not set they should be not shared.

Reply:
@tillkamppeter , I am facing trouble while testing the code. When I try to modify the attribute
printer-is-shared of the queue on the client, it shows the error unauthorised. Even
when trying to change the printer-is-shared bit from web interface, the error is
`Cannot change printer-is-shared for remote queues`. I really want to solve this issue but I wanted your help to figure out why remote queue, instead of local queue is appearing on the client side. On the `cups-browsed.conf` file on the client, the following options are set.
BrowseRemoteProtocols dnssd cups
BrowseInterval 60
BrowseTimeout 300
BrowsePoll 192.168.43.210:631
OnlyUnsupportedByCUPS Yes

Reply:
What is the device URI of the local queue with which you get the error (run `lpstat -v`)? Does it have a PPD file in `/etc/cups/ppd/`?

Reply:
The Device UR, which appears on the client is `implicitclass://D1_172_20_10_7/` where D1 is the description of the printer on the server. There also exists a ppd file for this printer.

Reply:
@tillkamppeter , I think my issue might be related to [this issue](https://github.com/apple/cups/issues/4738).

Reply:
CUPS actually does not allow re-sharing print queues which point to a remote CUPS queue. CUPS recognizes these queues in two ways (search for `|= CUPS_PRINTER_REMOTE` in the `scheduler/printers.c` file of the CUPS source code):
- The device URI is CUPS-typical, like for example `ipp://host:631/printers/queue` (resource is `/printers/queue`, not `/ipp/print` or `/ipp/print/queue`) or `ipp://service_name._ipp._tcp.local/cups` (resource `/cups` instead of no resource).
- The PPD file contains `*APRemoteQueueID: ...`.

Long time ago I followed the CUPS functionality by letting cups-browsed add a `*APRemoteQueueID: ...` line to the PPD. Simply search for "`*APRemoteQueueID:`" in `utils/cups-browsed.c` and comment out these lines for your further studies. Then restart cups-browsed. Now you should be able to share such a queue.
Note that all queues discovered by BrowsePoll are remote CUPS queues, so the cups-browsed as it is now will always add `*APRemoteQueueID: ...` to the PPD file.
There are two solutions: Either remove the code for adding `*APRemoteQueueID: ...` altogether or add a second boolean directive to `cups-browsed.conf`, call it "AllowResharingRemoteCUPSPrinters" and if set to "Yes" let the `*APRemoteQueueID: ...` not be added. Your directive will only take effect then.

Reply:
> add a second boolean directive to `cups-browsed.conf`, call it "AllowResharingRemoteCUPSPrinters" and if set to "Yes" let the `*APRemoteQueueID: ...` not be added.

@tillkamppeter , In that case, I've observed that print queues are automatically shared if `AllowResharingRemoteCUPSPrinters` is set to yes, may be because the option settings are copied of the shared printer. Therefore, there wouldn't be a need of the `NewBrowsePollQueuesShared` directive. But should I add it in case someone does not want to share the queues browsepolled from a server?